### ISSUE TITLE
Add a set of simulated datasets that display temporal variability in gammapy data

### ISSUE THREAD

**Issue by cgalelli** [gammapy/gammapy#4641]
It would be useful to have in `gammapy_data` a collection of simulated data that display some form of temporal variability; use cases could be the creation of tutorials and especially the validation of time-domain analyses and features. Currently, most time-domain tutorials and recipes rely on custom simulations that often have to be performed at every run or the limited data releases, and having a standardized simulation library would save time and improve efficiency.

The main discussion points would be:

- How many blocks of simulations to offer and which temporal behavior should they show: red noise, an asymmetrical Gaussian flare, a pulsating source, a combined long-term linear increase/decrease plus short-term bursts come to mind as possibilities. 
- Which data level should be provided: a first idea would be observations (also to keep consistency with the `lightcurve_simulation` tutorial that simulates data at this level).
- How big should the simulations be: a good balance should be struck between a significant dataset and a light load on the storage space. This of course changes with the different scenarios: for a short-term burst one observation night could be enough, but not for a long-term study.

The simulations would be provided with the gammapy scripts used to create them so that they could be updated in the future if needed.

**Comment by AtreyeeS** [comment]
Thanks @cgalelli for bringing this up. I guess it is difficult to plan in advance all the uses cases. . I thinks there are various methods of handling the noise, etc, and it depends whether gammapy recipes can be an entry point into detailed temporal modelling, or just shows how to interface gammapy with other analysis packages like stingray etc.

Is the physics case really important here? Or, can you have a very bright flare on an arbitrary scale?

I do not know if simulated observations are required. For most use cases, simulated DL4 datasets suffice. Do you have a link to some preliminary repo that we can look at to think a bit?

**Comment by cgalelli** [comment]
Hi @AtreyeeS and thank you very much for your interest!
 
I was proposing to provide observations to then be freer in the possible direct use cases, as in then the user can build the dataset however they like, but at least the starting fake observations are standardized. How do you feel about this? However, I agree that in general DL4 should be enough for most cases.

To be honest, I am torn on the physics case importance. I am having some conversations here in Meudon and I hope to soon have a clearer idea of the "physically realistic" vs "statistically interesting" debate. 

As soon as I have a preliminary repository I will share it here.

**Comment by cgalelli** [comment]
[Here](https://drive.google.com/file/d/1QmqY6OXud1uB4rAPbkZrbc2IzM0uQ7SV/view?usp=drive_link) is my proposal on the simulations and how to present the simulated datasets: the script shows the simulation process, using the `MapDatasetEventSampler`, and a quick peek at the simulated events.

I would love to have some feedback on the general algorithm and in particular on the handling of time axes and GTIs

**Comment by AtreyeeS** [comment]
Hi @cgalelli ! The notebook you link to basically follows https://docs.gammapy.org/1.1/tutorials/analysis-3d/event_sampling_nrg_depend_models.html#sphx-glr-tutorials-analysis-3d-event-sampling-nrg-depend-models-py, no?

**Comment by bkhelifi** [comment]
The idea to have a set of simulated datasets in DL3 will be very practical. It will allow to separate the DL3 generation (and the notebook that you have quoted, @AtreyeeS, contains indeed some basic ingredients to make it) with the Time Domain tutorials illustrating the tools... Without this separation, the tutorial will be much too long! This is why I support this idea...

Indeed, the big question is which time behaviour we wish to have in order to have a good tutorial. The astrophysical variety of temporal behaviour is almost infinite. But, here, the goal is to have good test cases for illustration of our functions.
Typically, there are 'behaviour':
- a flare as for AGNs: (log-normal) flux variation with changes of index
- a burst-like event (AGN, PWN, GRB): one can imagine a very very fast rising time with a 'power-law' decreasing time
- a red noise flux variation
- a white noise flux variation
With such variety of behaviours, one can have good test cases. Maybe I miss some temporal variety that one could add. In any case, such library will be useful for us: test functions, tutorials, maybe the benchmark.

**Comment by cgalelli** [comment]
> Hi @cgalelli ! The notebook you link to basically follows https://docs.gammapy.org/1.1/tutorials/analysis-3d/event_sampling_nrg_depend_models.html#sphx-glr-tutorials-analysis-3d-event-sampling-nrg-depend-models-py, no?

Hi @AtreyeeS and thanks!
Yes, it is a simple adaptation of that. It is only a bit more complex in the initial setup of the temporal behavior. I think that showing simulation algorithms that follow what we show in the tutorial is also good for "consistency".
My main doubt is whether I need to present or add some other set of tools or parameters for this particular case, for example, specific handling of GTIs.

**Comment by cgalelli** [comment]
> The idea to have a set of simulated datasets in DL3 will be very practical. It will allow to separate the DL3 generation (and the notebook that you have quoted, @AtreyeeS, contains indeed some basic ingredients to make it) with the Time Domain tutorials illustrating the tools... Without this separation, the tutorial will be much too long! This is why I support this idea...
> 
> Indeed, the big question is which time behaviour we wish to have in order to have a good tutorial. The astrophysical variety of temporal behaviour is almost infinite. But, here, the goal is to have good test cases for illustration of our functions. Typically, there are 'behaviour':
> 
> * a flare as for AGNs: (log-normal) flux variation with changes of index
> * a burst-like event (AGN, PWN, GRB): one can imagine a very very fast rising time with a 'power-law' decreasing time
> * a red noise flux variation
> * a white noise flux variation
>   With such variety of behaviours, one can have good test cases. Maybe I miss some temporal variety that one could add. In any case, such library will be useful for us: test functions, tutorials, maybe the benchmark.

Thanks @bkhelifi,
Indeed all the possibilities that you mentioned are in my first-uses list. As soon as we can reach a consensus on the final structure of the events and algorithm to present, I will implement the specific behaviours.

**Comment by AtreyeeS** [comment]
I don't know how standard the statistical tools are for analysing time series with gaps in the GTI.
If you have some specific  ideas, it would be very interesting to simulate such data (your GTI table will have multiple rows in that case).

Otherwise, starting with investigating the use cases @bkhelifi listed would be useful for the time being, and light curves with unequal / missing time bins can be  studied later

**Comment by Astro-Kirsty** [comment]
Is there any update for this work?

**Comment by cgalelli** [comment]
Yes and no: in essence, all the necessary things for these kinds of simulations and analyses are now in place in `gammapy_SyLC`, including possible techniques to tackle gaps in the GTIs and holes in data... In practice, I had completely forgotten the possibility of adding some sample simulations to gammapy data. I am open to either working on it if we decide that it's still relevant, or closing the issue if we are satisfied that the possibility of quickly simulating stuff using `SyLC` or any equivalent tool is enough.
